lib/timeout.rb

```diff
@@ -44,12 +44,43 @@ def self.handle_timeout(message) # :nodoc:
   end
 
   # :stopdoc:
-  CONDVAR = ConditionVariable.new
-  QUEUE = Queue.new
-  QUEUE_MUTEX = Mutex.new
-  TIMEOUT_THREAD_MUTEX = Mutex.new
-  @timeout_thread = nil
-  private_constant :CONDVAR, :QUEUE, :QUEUE_MUTEX, :TIMEOUT_THREAD_MUTEX
+  DATA = Struct.new(*%i{condvar queue queue_mutex timeout_thread_mutex timeout_thread})
+  private_constant :DATA
+
+  private module_function def make_data
```

Comment:
`private module_function` doesn't work on 2.6 and also doesn't give the right visibility, the singleton method is still public:
```
$ ruby -Ilib -rtimeout -e 'p Timeout.singleton_class.public_instance_methods(false)'
[:get_data, :ensure_timeout_thread_created, :timeout, :make_data]
```
It should be just `[:timeout]`, I'll add a test on master and fix `ensure_timeout_thread_created`